interface/patient_file/summary/labdata.php

```diff
@@ -79,40 +80,57 @@
 
 <script>
 function checkAll(bx) {
-    for (var tbls=document.getElementsByTagName("table"), i=tbls.length; i--; )
-      for (var bxs=tbls[i].getElementsByTagName("input"), j=bxs.length; j--; )
-         if (bxs[j].type=="checkbox")
+    for (var tbls=document.getElementsByTagName("table"), i=tbls.length; i--; ) {
+        for (var bxs=tbls[i].getElementsByTagName("input"), j=bxs.length; j--; ) {
+        if (bxs[j].type=="checkbox") {
             bxs[j].checked = bx.checked;
+        }
+      }
+    }
 }
-
 </script>
-<?php ##############################################################################
-echo "</head><body class='body_top'>";
-echo "<div id='labdata'>";
-echo "<h2>" . xlt('Labs') . "</h2>";
-echo "<span class='text'>";
-// some patient data...
-$spell  = "SELECT * ";
-$spell .= "FROM patient_data ";
-$spell .= "WHERE pid = ?";
-//---
-$myrow = sqlQuery($spell, array($pid));
-$lastname = $myrow["lname"];
-$firstname  = $myrow["fname"];
-$DOB  = $myrow["DOB"];
-
-
-
-if ($printable) {
-    echo "<table border=0>";
-    echo "<tr><td>" . xlt('Patient') . ": </td><td><b>" . text($lastname) . ", " . text($firstname) . "</b></td></tr>";
-    echo "<tr><td>" . xlt('Patient ID') . ": </td><td>" . text($pid) . "</td></tr>";
-    echo "<tr><td>" . xlt('Date of birth') . ": </td><td>" . text($DOB) . "</td></tr>";
-    echo "<tr><td>" . xlt('Access date') . ": </td><td>" . text(date('Y-m-d - H:i:s')) . "</td></tr>";
-    echo "</table>";
-}
-
+</head>
+<body>
+    <div id='labdata'>
+    <h2><?php echo xlt('Labs'); ?></h2>
+        <span class='text'>
```

Comment:
Hello @bradymiller  and @tywrenn 

Does OpenEMR have any guide to limit the use of the echo to render the HTML?

Such as this PR, the page consists of use echo to render the HTML view, but it caused hard to refine and maintain the HTML  code, So I would like to ask you, should I refactor to pure HTML?

Reply:
We don't. Since both ways work (yet might impact execution time if you use echo), I generally would allow both methods. If you want to put the extra work to converting it to HTML with the translations echoing only you can do that but there really is no difference.

Reply:
Okay, thanks for your explanation. I will convert a part of the echo into HTML.